src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+        label="httpPostBindingAuthnRequest"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.postBindingLogout"
+        label="httpPostBindingLogout"
+        isReadOnly={readOnly}
+      />
+
+      <SwitchField
+        field="config.wantAuthnRequestsSigned"
+        label="wantAuthnRequestsSigned"
+        isReadOnly={readOnly}
+      />
+
+      {wantAuthnSigned === "true" && (
+        <>
+          <FormGroup
+            label={t("signatureAlgorithm")}
+            labelIcon={
+              <HelpItem
+                helpText={th("signatureAlgorithm")}
+                forLabel={t("signatureAlgorithm")}
+                forID="kc-signatureAlgorithm"
+              />
+            }
+            fieldId="kc-signatureAlgorithm"
+          >
+            <Controller
+              name="config.signatureAlgorithm"
+              defaultValue="RSA_SHA256"
+              control={control}
+              render={({ onChange, value }) => (
+                <Select
+                  toggleId="kc-signatureAlgorithm"
+                  onToggle={() =>
+                    setSignatureAlgorithmDropdownOpen(
+                      !signatureAlgorithmDropdownOpen
+                    )
+                  }
+                  isOpen={signatureAlgorithmDropdownOpen}
+                  onSelect={(_, value) => {
+                    onChange(value as string);
+                    setSignatureAlgorithmDropdownOpen(false);
+                  }}
+                  selections={value}
+                  variant={SelectVariant.single}
+                >
+                  <SelectOption key={0} value="RSA_SHA1" />
+                  <SelectOption key={1} value="RSA_SHA256" isPlaceholder />
+                  <SelectOption key={2} value="RSA_SHA256_MGF1" />
+                  <SelectOption key={3} value="RSA_SHA512" />
+                  <SelectOption key={4} value="RSA_SHA512_MGF1" />
+                  <SelectOption key={5} value="DSA_SHA1" />
+                </Select>
+              )}
+            ></Controller>
+          </FormGroup>
+          <FormGroup
+            label={t("samlSignatureKeyName")}
+            labelIcon={
+              <HelpItem
+                helpText={th("samlSignatureKeyName")}
+                forLabel={t("samlSignatureKeyName")}
+                forID="kc-samlSignatureKeyName"
+              />
+            }
+            fieldId="kc-samlSignatureKeyName"
+          >
+            <Controller
+              name="config.xmlSigKeyInfoKeyNameTransformer"
+              defaultValue="keyID-option"
+              control={control}
+              render={({ onChange, value }) => (
+                <Select
+                  toggleId="kc-samlSignatureKeyName"
+                  onToggle={() =>
+                    setSamlSignatureKeyNameDropdownOpen(
+                      !samlSignatureKeyNameDropdownOpen
+                    )
+                  }
```

Comment:
There is no need to manage this state ourselves as the callback provides an argument for it.
```suggestion
                  onToggle={(isExpanded) =>
                    setSamlSignatureKeyNameDropdownOpen(isExpanded)
                  }
```

Reply:
fixed, ty